### ISSUE TITLE
Different Wayland Qt plugin is used in BiT (root) vs. BiT user-space

### ISSUE THREAD

**Issue by aryoda** [bit-team/backintime#1350]
## Background

The display manager support (`X11`, `Wayland`) of an app with a GUI is the responsibility of the app developer.

Ubuntu installs `XWayland` by default if the user installs Wayland which means all apps that do not support Wayland use the X11 API of XWayland which (partially) uses Wayland internally.

We are using Qt5 in BiT which supports X11. Qt5 Wayland support requires installing the `qtwayland5` package (on Ubuntu)

To use Wayland in Qt5 apps **additionally** the environment variable `QT_QPA_ARCHITECTURE` must be set to specific value
(`wayland-egl`).

If we do not explicitly configure the BiT launcher script to use Wayland BiT uses X11 (via XWayland on desktop environments with Wayland as display manager).

## Bug

We do set this env var ONLY in the BiT (root) launcher script
    https://github.com/bit-team/backintime/blob/49db33e62eaa8501120d85ffcaa6443492df7422/qt/backintime-qt_polkit#L1-L9
but not in the BiT (userland) launcher script
    https://github.com/bit-team/backintime/blob/49db33e62eaa8501120d85ffcaa6443492df7422/qt/backintime-qt#L22-L31
so only BiT (root) uses Wayland.

This also explains why we do not see userland BiT tickets for Wayland in our issues normally.

## Affected Bit versions

Current version of BiT (1.3.2) and most probably many older versions

## Expected behavior

- Both BiT versions should consistently use either X11 or Wayland on the same system
- Both BiT versions should use the same Wayland Qt plugin on Wayland.
- Prefer Wayland over X11 if Wayland is the active display manager

## (Partially) related issues and PRs

- #836
- #1349
- #1347 (not merged since the proposal in #836 is the more secure way of implementing BiT root but it could be a good starting point to use this work-around)

**Comment by aryoda** [comment]
FYI: I have added a `logger.debug` output to track the active Qt5 platform plugin (which determines if X11 or wayland is used):

```
> backintime-qt --debug
DEBUG: [qt/qttools.py:186 createQApplication] QT QPA platform plugin: xcb
```

The interpretation of the plugin name will be described in my upcomding "BiT developer's doc for wayland" (in a few days)...

**Comment by aryoda** [comment]
Upcoming fix is described here: https://github.com/bit-team/backintime/issues/836#issuecomment-1875081695

**Comment by aryoda** [comment]
Closed with PR #1597

**Comment by aryoda** [comment]
Closed with PR #1597
